src/App.js

```diff
@@ -1,16 +1,57 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+
+// const list = []
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+
+constructor() {
+  super();
+  this.state = {
+    list: [],
+    inputValue: '',
+  };
+}
+
+addTodo = e => {
+  e.preventDefault();
+  const newTodo = {
+    id: Date.now(),
```

Comment:
Notice how you are creating an id for each one of your todo Items? This would make a much better key for your map, because it insures your key is always going to be pointed to the same todo. indexes can change